### PR TITLE
Customer module

### PR DESCRIPTION
**Pull request by sgreddy123**
Customer Module

### CODE REVIEW

**Review by SalukulaMahendra — APPROVED** [review]
no conflicts

File: SQL/scripts.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+create schema 2024_batch_jdbc;
+use 2024_batch_jdbc;
+	create table customer(id int,name varchar(70), address varchar(100), phone_Num long);
+	select * from customer;
```
Comment (khadri2021) [inline SQL/scripts.sql]: why we need select query???

File: src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerInsertDao.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Scanner;
+
+import com.khadri.jdbc.prepare.data.customer.dao.model.Customer;
+
+public class CustomerInsertDao {
+	public static void insertData(Customer cust) throws Exception {
+		// write JDBC logic here
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerInsertDao.java]: why this comment required
?

File: src/main/java/com/khadri/jdbc/prepare/data/customer/dao/model/Customer.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package com.khadri.jdbc.prepare.data.customer.dao.model;
+
+public class Customer {
+	private int id;
+	private String name;
+	private String address;
+	private long phone_num;
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/customer/dao/model/Customer.java]: where is the camelcase for property?

File: src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerInsertDao.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+		Scanner sc = new Scanner(System.in);
+		Connection con = null;
+		PreparedStatement pstmt =null;
+
+		try {
+			Class.forName("com.mysql.cj.jdbc.Driver");
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerInsertDao.java]: creating connection is common logic for everyone, create a separate class  and prepare connection object in method call that method here to use the connection object.

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 5]
```diff
@@ -11,12 +14,16 @@ public class JdbcDataDriver {
 
 	private Scanner scanner;
 	private StudentInsertDao insertDao;
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: define like stdDao

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 6]
```diff
@@ -11,12 +14,16 @@ public class JdbcDataDriver {
...
+	//private Customer cust;
 	{
 		scanner = new Scanner(System.in);
 		insertDao = new StudentInsertDao();
+		insertDao2= new CustomerInsertDao();
+		
 	}
 
-	public static void main(String[] args) {
+	public static void main(String[] args) throws Exception {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: which exception is throwing here, do we need this throws Exception??

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 7]
```diff
@@ -28,21 +35,34 @@ public static void main(String[] args) {
...
 			StudentDataProcessor stdProcessor = new StudentDataProcessor(scanner,insertDao);
 
 			for (int i = 0; i < recordCount; i++) {
 				stdProcessor.process(dataProcessId);
 			}
+			break;
+		case 3:
+			System.out.println("How many records you want to insert ? : ");
+			int recordCount2 = scanner.nextInt();
+			//scanner.nextLine();
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: if not using remove commented lines code

**Comment by sgreddy123** [comment]
ok sir i will modify and i will commit it back

**Review by manvithagali52 — APPROVED** [review]

File: src/main/java/com/khadri/jdbc/prepare/data/customer/dao/model/Customer.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
+package com.khadri.jdbc.prepare.data.customer.dao.model;
+
+public class Customer {
+	private int id;
+	private String name;
+	private String address;
+	private long phone_Num;
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/customer/dao/model/Customer.java]: why you need separator?

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 9]
```diff
@@ -3,17 +3,22 @@
...
 import com.khadri.jdbc.prepare.data.student.dao.StudentInsertDao;
 import com.khadri.jdbc.prepare.data.student.processor.StudentDataProcessor;
 
 public class JdbcDataDriver {
 
-	private Scanner scanner;
-	private StudentInsertDao insertDao;
+	public Scanner scanner;
+	public StudentInsertDao insertDao;
+	public CustomerInsertDao CustDao;
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: it should be camel case not titled case

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 10]
```diff
@@ -25,24 +30,41 @@ public static void main(String[] args) {
 		});
 
 		JdbcDataDriver csvDataDriver = new JdbcDataDriver();
+
 		csvDataDriver.process();
+
 	}
 
-	private void process() {
+	public void process() {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: revert to private.

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 11]
```diff
@@ -25,24 +30,41 @@ public static void main(String[] args) {
...
 
-			StudentDataProcessor stdProcessor = new StudentDataProcessor(scanner,insertDao);
+			StudentDataProcessor stdProcessor = new StudentDataProcessor(scanner, insertDao);
 
 			for (int i = 0; i < recordCount; i++) {
 				stdProcessor.process(dataProcessId);
 			}
+			break;
+		case 3:
+			System.out.println("How many records you want to insert ? : ");
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: maintain same sentence.

Comment (sgreddy123) [reply src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: i modified sir

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java (hunk 10 above)
Comment (sgreddy123) [reply src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: i reverted  public to private sir

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java (hunk 9 above)
Comment (sgreddy123) [reply src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: i followed camel case sir

File: src/main/java/com/khadri/jdbc/prepare/data/customer/dao/model/Customer.java (hunk 8 above)
Comment (sgreddy123) [reply src/main/java/com/khadri/jdbc/prepare/data/customer/dao/model/Customer.java]: i removed sir

**Pull request merged**